modules/datepicker/basepicker.tsx

```diff
@@ -0,0 +1,83 @@
+import * as React from 'react'
+import {Button} from 'react-native'
+
+import {useTheme} from '@frogpond/app-theme'
+import type {AppTheme} from '@frogpond/app-theme'
+
+import moment from 'moment-timezone'
+import type {Moment} from 'moment-timezone'
+
+import DateTimePicker from '@react-native-community/datetimepicker'
+import type {Event} from '@react-native-community/datetimepicker'
+
+import {BaseDatetimePickerProps} from './types'
+
+const FORMATS = {
+	date: 'YYYY-MM-DD',
+	datetime: 'YYYY-MM-DD HH:mm',
+	time: 'HH:mm',
+}
+
+export const BaseDateTimePicker = (
+	props: BaseDatetimePickerProps,
+): JSX.Element => {
```

Comment:
Is a `JSX.Element` the right thing to return here? Or should we return a `React.ReactNode`? (I'm not super familiar with the standard, so this is more just a question.)

Reply:
This is a pretty [comprehensive answer](https://stackoverflow.com/a/59840095) which I'll be quoting a section of.

> Why do the render methods of class components return `ReactNode`, and function components return `ReactElement`.
> 
> It's a TS type incompatibility.
> 
> * TS class component returns `ReactNode` with `render()`
> * TS function component returns `JSXElement | null`